rectools/models/rank/rank_implicit.py

```diff
@@ -205,16 +203,16 @@ def rank(  # pylint: disable=too-many-branches
             Whitelist of object ids.
             If given, only these items will be used for recommendations.
             Otherwise all items from dataset will be used.
-        num_threads : int, default 0
-            Will be used as `num_threads` parameter for `implicit.cpu.topk.topk`. Omitted if use_gpu is True
-        use_gpu : bool, default False
-            If True `implicit.gpu.KnnQuery().topk` will be used instead of classic cpu version.
 
         Returns
         -------
         (InternalIds, InternalIds, Scores)
             Array of subject ids, array of recommended items, sorted by score descending and array of scores.
         """
+        if filter_pairs_csr is not None and filter_pairs_csr.shape[0] != len(subject_ids):
+            explanation = "assumed that filter_pairs_csr and subject_ids are aligned"
+            raise ValueError(explanation)
```

Comment:
Could you please add a test for this line?

Reply:
done